packages/jmix-react-ui/src/ui/table/DataTable.test.tsx

```diff
@@ -24,29 +28,78 @@ window.matchMedia = () => ({
-                       {manufacturer: 'BBB', model: '002', id: '2'},
-                     ]}
-                     count={2}
-          />
-        </IntlProvider>
-      </Provider>
-    );
-    const table = await screen.findByRole('table');
-    expect(table).toBeInTheDocument();
+  const dataTableTestRenderer = create(
+    <Provider mainStore={mainStore}>
+      <IntlProvider locale='en'>
+        <DataTable loading={false}
+                   onFilterChange={noop}
+                   onSortOrderChange={noop}
+                   onPaginationChange={noop}
+                   entityName={'scr_Car'}
+                   columnDefinitions={[
+                     'manufacturer',
+                     'model'
+                   ]}
+                   items={[
+                     {manufacturer: 'AAA', model: '001', id: '1'},
+                     {manufacturer: 'BBB', model: '002', id: '2'},
+                   ]}
+                   count={2}
+        />
+      </IntlProvider>
+    </Provider>
+  );
+
+  const dataTableTestInstance = dataTableTestRenderer.root;
+
+  const intlComponent = dataTableTestInstance.children[0] as ReactTestInstance;
+  const mainStoreComponent = intlComponent.children[0] as ReactTestInstance;
+  const metadataComponent = mainStoreComponent.children[0] as ReactTestInstance;
+  const observerComponent = metadataComponent.children[0] as ReactTestInstance;
+  const dataTableComponent = observerComponent.children[0] as ReactTestInstance;
+
+
+  it('renders',  () => {
+    expect(dataTableTestInstance).not.toBeNull();
   });
+
+  it('wrapper has a DataTable child that has a selectedRowKeys property', () => {
+    expect(dataTableComponent.instance.selectedRowKeys).not.toBeUndefined();
+  })
+
+  it('selectedRowKeys must be an empty array', () => {
+    expect(dataTableComponent.instance.selectedRowKeys).toBeInstanceOf(Array);
+    expect(dataTableComponent.instance.selectedRowKeys).toHaveLength(0);
+  })
+
+  it('DataTable.onRowSelectionChange must be called once after click on the row', async () => {
+    // const onRowSelectionChangeSpy = jest.spyOn(dataTableComponent.instance, 'onRowSelectionChange')
+
+    // Find the rows
+    const rows = dataTableComponent.findAllByProps({className: 'ant-table-row ant-table-row-level-0'})
+
+    expect(dataTableComponent.instance.disposers.length).toBeGreaterThanOrEqual(1);
+    const prevSelectedRowKeys = [...dataTableComponent.instance.selectedRowKeys];
+    expect(prevSelectedRowKeys).toHaveLength(0);
+
+    rows[0].props.onClick(new Event('click'))
+    const newSelectedRowKeys = dataTableComponent.instance.selectedRowKeys;
+    expect(prevSelectedRowKeys).not.toEqual(newSelectedRowKeys);
+    expect(newSelectedRowKeys).toHaveLength(1);
+
+    await wait(1000);
+
+    // ??? Why it isn't working? TODO: think a little about it later
+    // expect(onRowSelectionChangeSpy).toBeCalled();
+    // expect(onRowSelectionChangeSpy).toHaveBeenCalledTimes(1);
```

Comment:
Let's remove the call to `wait` since we don't assert anything afterwards.

Reply:
Also, using timeouts like this can lead to fragile tests (it works locally, then *sometimes* it fails on CI, etc.)

`@testing-library/react` has some [async utilities](https://testing-library.com/docs/dom-testing-library/api-async/) that are a more robust way to wait for something.